src/board.rs

```diff
@@ -45,13 +42,117 @@ impl Chessboard {
+            white_queen: 0b0000000000000000000000000000000000000000000000000000000000000000,
+            white_rooks: 0b0000000000000000000000000000000000000000000000000000000000000000,
+            black_pawns: 0b0000000000000000000000000000000000000000000000000000000000000000,
+            black_knights: 0b0000000000000000000000000000000000000000000000000000000000000000,
+            black_bishops: 0b0000000000000000000000000000000000000000000000000000000000000000,
+            black_king: 0b0000000000000000000000000000000000000000000000000000000000000000,
+            black_queen: 0b0000000000000000000000000000000000000000000000000000000000000000,
+            black_rooks: 0b0000000000000000000000000000000000000000000000000000000000000000,
+            castling_rights: 0b1111,
+            en_passant: 0,
+            white_turn: true,
+        }
+    }
+
+    /// Forsyth–Edwards Notation Parser
+    /// * `fen` - The FEN to be converted to a Chessboard.
+    /// # Return: Chessboard with the position from the FEN.
+    pub fn from_string(fen: &str) -> Chessboard {
+        let mut chessboard = Chessboard::new();
+
+        // Split the FEN string into parts using ' ' as the delimiter
+        let fen_parts: Vec<&str> = fen.split_whitespace().collect();
+
+        // Parse the piece placement part of the FEN string
+        let board_rows: Vec<&str> = fen_parts[0].split('/').collect();
+        for (mut rank, row) in board_rows.iter().rev().enumerate() {
+            rank += 1;
+            let mut file = 0; // Initialize the file (column) index
+            for piece in row.chars() {
+                if piece.is_digit(10) {
+                    let empty_squares = piece.to_digit(10).unwrap() as usize;
+                    file += empty_squares; // Skip empty squares
+                } else {
+                    let square_index = 8 * (rank - 1) + file;
+                    // Update the corresponding bitboard based on the piece type and color
+                    match piece {
+                        'p' => chessboard.black_pawns |= pow(2, square_index),
+                        'r' => chessboard.black_rooks |= pow(2, square_index),
+                        'b' => chessboard.black_bishops |= pow(2, square_index),
+                        'k' => chessboard.black_king |= pow(2, square_index),
+                        'q' => chessboard.black_queen |= pow(2, square_index),
+                        'n' => chessboard.black_knights |= pow(2, square_index),
+                        'P' => chessboard.white_pawns |= pow(2, square_index),
+                        'R' => chessboard.white_rooks |= pow(2, square_index),
+                        'B' => chessboard.white_bishops |= pow(2, square_index),
+                        'K' => chessboard.white_king |= pow(2, square_index),
+                        'Q' => chessboard.white_queen |= pow(2, square_index),
+                        'N' => chessboard.white_knights |= pow(2, square_index),
```

Comment:
There is a builtin [pow function](https://doc.rust-lang.org/std/?search=pow) in rust. That would be great if we could use it here instead of importing a new module.
This might need to become it's own issue, as we might need to rework what datatype we use for `file` and `rank`